### PR TITLE
KFKA-4340: follow up patch for KAFAK-4340

### PR DESCRIPTION
**Pull request by becketqin**

### CODE REVIEW

**Comment by becketqin** [comment]
cc @ijuma

File: docs/upgrade.html [hunk 1]
```diff
@@ -77,6 +77,7 @@
     <li>Several new fields including "security.protocol", "connections.max.idle.ms", "retry.backoff.ms", "reconnect.backoff.ms" and "request.timeout.ms" were added to
         StreamsConfig class. User should pay attenntion to the default values and set these if needed. For more details please refer to <a href="#streamsconfigs">3.5 Kafka Streams Configs</a>.</li>
     <li>The <code>offsets.topic.replication.factor</code> broker config is now enforced upon auto topic creation. Internal auto topic creation will fail with a GROUP_COORDINATOR_NOT_AVAILABLE error until the cluster size meets this replication factor requirement.</li>
+    <li>The default value of <code>message.timestamp.difference.max.ms</code> has been changed to the same as <code>retention.ms</code></li>
```
Comment (junrao) [inline docs/upgrade.html:80]: It seems that we need to add that to a new section for upgrade_1030_notable?

File: core/src/test/scala/unit/kafka/log/LogTest.scala [hunk 2]
```diff
@@ -988,6 +988,16 @@ class LogTest extends JUnitSuite {
...
   }
+  
+  @Test
+  def testAppendWithNullTimestamp(): Unit = {
+    val log = new Log(logDir,
+      LogConfig(),
+      recoveryPoint = 0L,
+      time.scheduler,
+      time)
+    log.append(MemoryRecords.withRecords(Record.create(-1L, "key".getBytes, "value".getBytes)))
```
Comment (junrao) [inline core/src/test/scala/unit/kafka/log/LogTest.scala]: Instead of -1, could we just reference Record.NO_TIMESTAMP?

**Review by junrao — COMMENTED** [review]
@becketqin : Thanks for the patch. LGTM. Just a couple of minor comments.

**Comment by junrao** [comment]
Also, @ijuma brought up a good point. Right now, in KafkaConfig, if retention time is set, but MessageTimestampDifferenceMaxMs is not, MessageTimestampDifferenceMaxMs defaults to the default retention time. It seems that MessageTimestampDifferenceMaxMs should default to the customized retention time instead.

**Comment by cmccabe** [comment]
Thanks for the patch, @becketqin.  It looks good.

@ijuma, @junrao, do you think we'll be able to fix this soon to get the tests passing again, or should we revert KAFKA-4340 so that we have more time to think about it and get it right?  Right now trunk is broken.

**Comment by becketqin** [comment]
@cmccabe I'm working on the patch and will submit an update today. I am fine to revert the patch as it is not that critical to change the default configuration value.

**Review by ijuma — APPROVED** [review]
Thanks for the update, LGTM.

**Comment by ijuma** [comment]
Fixed the upgrade notes, tweaked the documentation of the config to mention the new default and merged to trunk.

**Pull request closed without merging**